src/shogun/base/SGObject.h

```diff
@@ -98,15 +99,12 @@ using stringToEnumMapType = std::unordered_map<std::string, std::unordered_map<s
 			this->m_gradient_parameters->add(param, name, description);        \
 	}
 
-#define SG_ADD5(param, name, description, param_properties, auto_init)         \
+#define SG_ADD5(param, name, description, param_properties, auto_or_constraint)\
```

Comment:
@karlnapf the issue with this is that now auto_init and constraint are mutually exclusive. This is fine for now, but will have to be taken into consideration when switching to a function, instead of macro.